### ISSUE TITLE
A subgraph in a subgraph doesn't highlight and can't be opened with double click

### ISSUE THREAD

**Issue by ElementalAndy** [OurMachinery/themachinery-public#1072]
- **Version:** Git Master Branch
- **OS:** Windows

I have an entity with a graph component. In that root graph I place an instance of a subgraph prototype. I get the nice highlighting when I simulate telling me the subgraph is getting it's graph events. If I now put another instance of a subgraph prototype in the first one the highlighting doesn't work for the second subgraph. I can't interact with it either i.e. double clicking on it with the left mouse button does nothing.

I can now open the prototype of the first subgraph and then double click the second one to open the instance of that one. But at this point there seems to be confusion regarding which graph instance I have opened. This isn't the simulating instance of the second subgraph and the combo box at the top right of the Entity Graph tab says "Choose running instance".

Below is the first subgraph instance that works as expected.

![trigger_subgraph_test](https://user-images.githubusercontent.com/91697782/154667623-c77de894-af40-4934-8232-4f81724f5a35.gif)

Below is the opened first subgraph instance. Here the highlighting doesn't work for the second subgraph node.  

![trigger_subsubgraph_instance_test](https://user-images.githubusercontent.com/91697782/154667001-9c8c784c-5ace-467f-add5-9a3e1496c23d.gif)

**Comment by ElementalAndy** [comment]
tldr;

It's difficult to debug subgraphs in subgraphs :)

**Comment by simon-ourmachinery** [comment]
Suspect: It could be that the graph compiler is not telling the debugger where the flattened subgraph is.

**Comment by simon-ourmachinery** [comment]
This is the issue with the @karl-zylinski-ourmachinery  subgraph in subgraph instance not being able to open on double click.

**Comment by karl-zylinski-ourmachinery** [comment]
@ElementalAndy I am currently working on making it possible to open inherited things such as graphs and entities and components so one can inspect them without first having to override them.

@simon-ourmachinery I think https://github.com/OurMachinery/themachinery-public/issues/1073 is similar enough to the cant-open-inhertied-subgraph problem, I will fix them both at the same time.

**Comment by ElementalAndy** [comment]
@karl-zylinski-ourmachinery @simon-ourmachinery Thanks, I'm definitely guilty of not searching for already entered issues this time 😸
